kurl_util/cmd/installermerge/main.go

```diff
@@ -0,0 +1,233 @@
+			mergedConfig[key] = oldVal
+			continue
+		}
+
+		if !oldOk && newOk {
+			mergedConfig[key] = newVal
+			continue
+		}
+
+		oldValMap, isOldMap := oldVal.(map[string]interface{})
+		if !isOldMap {
+			mergedConfig[key] = newVal
+			continue
+		}
+
+		newValMap, isNewMap := newVal.(map[string]interface{})
+		if !isNewMap {
+			mergedConfig[key] = newVal
+			continue
+		}
+
+		mergedConfig[key] = mergeJSONMaps(oldValMap, newValMap)
+	}
+	return mergedConfig
+}
+
+func mergeKeys(config1 map[string]interface{}, config2 map[string]interface{}) []string {
+	mergedMap := make(map[string]struct{})
+	for key := range config1 {
+		mergedMap[key] = struct{}{}
+	}
+	for key := range config2 {
+		mergedMap[key] = struct{}{}
+	}
+
+	allKeys := make([]string, 0)
+	for key := range mergedMap {
+		allKeys = append(allKeys, key)
+	}
+
+	return allKeys
+}
+
+func mergeDockerConfigData(oldconfigdata []byte, newconfigdata []byte) ([]byte, error) {
+	oldconfigdata = bytes.TrimSpace(oldconfigdata)
+	newconfigdata = bytes.TrimSpace(newconfigdata)
+
+	if len(oldconfigdata) == 0 && len(newconfigdata) == 0 {
+		return nil, nil
+	}
+
+	if len(oldconfigdata) == 0 {
+		return newconfigdata, nil
+	}
+
+	if len(newconfigdata) == 0 {
+		return oldconfigdata, nil
+	}
+
+	oldconfig := make(map[string]interface{})
+	if err := json.Unmarshal(oldconfigdata, &oldconfig); err != nil {
+		return nil, errors.Wrap(err, "failed to parse existing config")
+	}
+
+	newconfig := make(map[string]interface{})
+	if err := json.Unmarshal(newconfigdata, &newconfig); err != nil {
+		return nil, errors.Wrap(err, "failed to parse new config")
+	}
+
+	mergedconfig := mergeJSONMaps(oldconfig, newconfig)
+
+	mergedconfigdata, err := json.MarshalIndent(mergedconfig, "", "  ")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal merged config")
+	}
+
+	return mergedconfigdata, nil
+}
+
+func mergeYamlConfigData(oldconfigdata []byte, newconfigdata []byte) ([]byte, error) {
+	oldconfigdata = bytes.TrimSpace(oldconfigdata)
+	newconfigdata = bytes.TrimSpace(newconfigdata)
+
+	if len(oldconfigdata) == 0 && len(newconfigdata) == 0 {
+		return nil, nil
+	}
+
+	if len(oldconfigdata) == 0 {
+		return newconfigdata, nil
+	}
+
+	if len(newconfigdata) == 0 {
+		return oldconfigdata, nil
+	}
+
+	oldconfig := make(map[string]interface{})
+	if err := yaml.Unmarshal(oldconfigdata, &oldconfig); err != nil {
+		return nil, errors.Wrap(err, "failed to parse existing config")
+	}
+
+	newconfig := make(map[string]interface{})
+	if err := yaml.Unmarshal(newconfigdata, &newconfig); err != nil {
+		return nil, errors.Wrap(err, "failed to parse new config")
+	}
+
+	mergedconfig := mergeYAMLMaps(oldconfig, newconfig)
+
+	mergedconfigdata, err := yaml.Marshal(mergedconfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal merged config")
+	}
+
+	return mergedconfigdata, nil
+}
+
+func main() {
+	kurlscheme.AddToScheme(scheme.Scheme)
+
+	oldInstaller, _ := getInstallerConfigFromYaml("old.yaml")
```

Comment:
Shouldn't these files come from command line arguments?  And also, error handling?

Reply:
the mergeConfigYamlData was really what were looking at, that will be done before its read for use in kurl